src/NuGet.Server.Core/Infrastructure/KnownPathUtility.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+
+namespace NuGet.Server.Core.Infrastructure
+{
+    public static class KnownPathUtility
+    {
+        /// <summary>
+        /// Determines if the file path could have been generated by <see cref="ExpandedPackageRepository"/>.
+        /// </summary>
+        /// <param name="path">The file path to parse.</param>
+        /// <param name="id">The package ID found.</param>
+        /// <param name="version">The package version found.</param>
+        /// <returns>True if the file name is known.</returns>
+        public static bool TryParseFileName(string path, out string id, out SemanticVersion version)
+        {
+            id = null;
+            version = null;
+
+            if (path == null)
+            {
+                return false;
+            }
+
+            var pathPieces = path.Split(Path.DirectorySeparatorChar);
+            if (pathPieces.Length < 3) // {id}\{version}\{file name}
```

Comment:
>pathPieces.Length < 3 [](start = 16, length = 21)

Should this be `!= 3`? What if I drop a package in a path like `blah\blah\blah\{id}\{version}\{file name}`?

Reply:
Yeah, that's a great point!

Reply:
No absolute paths then?

Reply:
Correct.